test/Microsoft.AspNetCore.SignalR.Tests.Utils/ServerFixture.cs

```diff
@@ -23,35 +26,34 @@ public class ServerFixture<TStartup> : IDisposable
         private IWebHost _host;
         private IApplicationLifetime _lifetime;
         private readonly IDisposable _logToken;
-        private AsyncForwardingLoggerProvider _asyncLoggerProvider;
+
+        private readonly LogSinkProvider _logSinkProvider;
 
         public string WebSocketsUrl => Url.Replace("http", "ws");
 
         public string Url { get; private set; }
 
         public ServerFixture()
         {
-            _asyncLoggerProvider = new AsyncForwardingLoggerProvider();
+            _logSinkProvider = new LogSinkProvider();
 
             var testLog = AssemblyTestLog.ForAssembly(typeof(TStartup).Assembly);
             _logToken = testLog.StartTestLog(null, $"{nameof(ServerFixture<TStartup>)}_{typeof(TStartup).Name}", out _loggerFactory, "ServerFixture");
-            _loggerFactory.AddProvider(_asyncLoggerProvider);
+            _loggerFactory.AddProvider(_logSinkProvider);
             _logger = _loggerFactory.CreateLogger<ServerFixture<TStartup>>();
-            // We're using 127.0.0.1 instead of localhost to ensure that we use IPV4 across different OSes
-            Url = "http://127.0.0.1:" + GetNextPort();
 
-            StartServer(Url);
+            StartServer();
         }
 
-        public void SetTestLoggerFactory(ILoggerFactory loggerFactory)
+        private void StartServer()
         {
-            _asyncLoggerProvider.SetLoggerFactory(loggerFactory);
-        }
+            // We're using 127.0.0.1 instead of localhost to ensure that we use IPV4 across different OSes
+            var url = "http://127.0.0.1:0";
 
-        private void StartServer(string url)
-        {
             _host = new WebHostBuilder()
-                .ConfigureLogging(builder => builder.AddProvider(new ForwardingLoggerProvider(_loggerFactory)))
+                .ConfigureLogging(builder => builder
+                    .AddProvider(_logSinkProvider)
```

Comment:
Doesn't this provider get added twice now?

Reply:
Yeah, but it actually isn't needed above. I'll remove it there.